### PR TITLE
Add schema makefile

### PR DESCRIPTION
**Pull request by mogottsch**
### What this PR does / why we need it

This PR adds a JSON schema-related makefile, which provides commands to work with `values.schema.json` and `values.yaml`.
Originally this change was introduced in [this PR](https://github.com/giantswarm/cluster-aws/pull/275/files), which I closed because I mistakenly thought that the changes were already merged.

I've also taken the liberty to add an explanation/instruction on how to work with the schema-related files in the readme.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,3 +195,44 @@ machinePools:
...
 ```
+
+
+## Maintaining `values.schema.json` and `values.yaml`
+
+**tldr**:  
+We only maintain `values.schema.json` and automatically generate `values.yaml` from it.
+```
+make schema-normalize
+make schmema-validate
```
Comment (tuladhar) [inline README.md:206]:
```suggestion
make schema-validate
```

File: README.md [hunk 2]
```diff
@@ -195,3 +195,44 @@ machinePools:
...
+- Validate whether the requirements for cluster app schemas are met
+- Check whether schema is normalized
+
+The first point can be achieved with:
+```
+make schema-normalize
+```
+The second to fourth point can be achieved with:
+```
+make schmema-validate
```
Comment (tuladhar) [inline README.md:229]:
```suggestion
make schema-validate
```

**Review by marians — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -195,3 +195,44 @@ machinePools:
...
+
+
+## Maintaining `values.schema.json` and `values.yaml`
+
+**tldr**:  
+We only maintain `values.schema.json` and automatically generate `values.yaml` from it.
+```
+make schema-normalize
+make schmema-validate
+make values-generate
```
Comment (tuladhar) [inline README.md:207]: Should it be `generate-values`?
```suggestion
make generate-values
```

File: README.md [hunk 4]
```diff
@@ -195,3 +195,44 @@ machinePools:
...
+```
+make schmema-validate
+```
+
+
+The JSON schema in `values.schema.json` should contain defaults defined with the `default` keyword.
+These defaults should be same as those defined in `values.yaml`. 
+This allows us to generate `values.yaml` from `values.schema.json` with:
+```
+make values-generate
```
Comment (tuladhar) [inline README.md:237]: Should it be `generate-values`?
```suggestion
make generate-values
```

File: README.md (hunk 1 above)
Comment (AndiDog) [reply README.md:206]: Actually, it's `normalize-schema` and `validate-schema` (word order; repeated below)

Comment (mogottsch) [reply README.md]: https://github.com/giantswarm/cluster-aws/pull/291/commits/b244918cc682f18ca837bff27b484f43e3aa55f6

File: README.md (hunk 2 above)
Comment (mogottsch) [reply README.md]: https://github.com/giantswarm/cluster-aws/pull/291/commits/b244918cc682f18ca837bff27b484f43e3aa55f6

File: README.md (hunk 3 above)
Comment (mogottsch) [reply README.md]: https://github.com/giantswarm/cluster-aws/pull/291/commits/b244918cc682f18ca837bff27b484f43e3aa55f6

File: README.md (hunk 4 above)
Comment (mogottsch) [reply README.md]: https://github.com/giantswarm/cluster-aws/pull/291/commits/b244918cc682f18ca837bff27b484f43e3aa55f6

**Comment by mogottsch** [comment]
> are these changes managed through [our github automation](https://github.com/giantswarm/github)?

The makefiles and the workflow files are. The readme part is not.

File: README.md [hunk 5]
```diff
@@ -195,3 +195,44 @@ machinePools:
...
+
+
+## Maintaining `values.schema.json` and `values.yaml`
+
+**tldr**:  
+We only maintain `values.schema.json` and automatically generate `values.yaml` from it.
+```
+make normalize-schema
+make validate-schema
+make generate-schema
```
[the remark this answers is not in the document]
Comment (mogottsch) [reply README.md:207]: damn I need a coffe

[the remark this answers is not in the document]
Comment (mogottsch) [reply README.md:207]: thanks for reading carefully

**Pull request merged**